src/native/util/os_linux.cpp

```diff
@@ -24,6 +26,13 @@ get_current_uid()
 const uid_t ThreadScope::orig_uid = getuid();
 const gid_t ThreadScope::orig_gid = getgid();
 
+const std::vector<gid_t> ThreadScope::orig_groups = [] {
```

Comment:
As discussed - add a comment here, or near the set/unset of the groups to explain why or what the code do what it does.

Reply:
Added